cmd/policy-assistant/pkg/cli/analyze.go

```diff
@@ -104,27 +107,37 @@ func RunAnalyzeCommand(args *AnalyzeArgs) {
 			kubeNamespaces = nsList.Items
 			namespaces = []string{v1.NamespaceAll}
 		}
-		kubePolicies, err = kube.ReadNetworkPoliciesFromKube(kubeClient, namespaces)
-		if err != nil {
+
+		//TODO: add a flag for the timeout
+		ctx, cancel := context.WithTimeout(context.TODO(), 15*time.Second)
+		defer cancel()
+
+		kubePolicies, kubeANPs, kubeBANPs, netErr, anpErr, banpErr = kube.ReadNetworkPoliciesFromKube(ctx, kubeClient, namespaces)
+
+		if netErr != nil {
 			logrus.Errorf("unable to read network policies from kube, ns '%s': %+v", namespaces, err)
 		}
-		kubePods, err = kube.GetPodsInNamespaces(kubeClient, namespaces)
-		if err != nil {
-			logrus.Errorf("unable to read pods from kube, ns '%s': %+v", namespaces, err)
+		if anpErr != nil {
+			fmt.Printf("Unable to fetch admin network policies: %s \n", anpErr)
+		}
+		if banpErr != nil {
+			fmt.Printf("Unable to fetch base admin network policies: %s \n", banpErr)
 		}
 	}
 	// 2. read policies from file
 	if args.PolicyPath != "" {
-		policiesFromPath, err := kube.ReadNetworkPoliciesFromPath(args.PolicyPath)
+		policiesFromPath, anpsFromPath, banpFromPath, err := kube.ReadNetworkPoliciesFromPath(args.PolicyPath)
 		utils.DoOrDie(err)
 		kubePolicies = append(kubePolicies, policiesFromPath...)
+		kubeANPs = append(kubeANPs, anpsFromPath...)
+		kubeBANPs = banpFromPath
```

Comment:
Probably a bit misleading name, but kubeBanps is a pointer to struct not slice. Since the whole thing is built around that there would be only one BANP, I'll keep it that way and rename it something more appropriate.

Reply:
Agh right. In that case, could we log a warning if banp was already set prior? I would probably keep the banp from API server over e.g. the example banp but that’s a pretty unique scenario. I’m ok with any approach as long as we log to inform the user of conflicting banp’s